### PR TITLE
chore: Update datafusion and arrow / parquet / arrow-flight 25.0.0

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
* keep up with dependencies

# Changes:
1. Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/feff5dc805449afa4a7ecb5ca88b979ad5739cce
2. Update to `arrow` / `parquet` / `arrow-flight` 25.0.0

### CODE REVIEW

File: iox_query/src/util.rs [hunk 1]
```diff
@@ -211,14 +211,10 @@ impl<'a> ExprRewriter for MissingColumnsToNull<'a> {
         // Until then, we need to know what type of expr the column is
         // being compared with, so workaround by finding the datatype of the other arg
         match expr {
-            Expr::BinaryExpr { left, op, right } => {
+            Expr::BinaryExpr(BinaryExpr { left, op, right }) => {
```
Comment (alamb) [inline iox_query/src/util.rs:214]: Due to https://github.com/apache/arrow-datafusion/pull/3835

File: predicate/src/delete_expr.rs [hunk 2]
```diff
@@ -173,33 +177,14 @@ mod tests {
...
-            left: Box::new(Expr::Column(datafusion::prelude::Column {
-                relation: None,
-                name: "foo".to_string(),
-            })),
-            op: datafusion::logical_expr::Operator::Eq,
-            right: Box::new(Expr::Literal(datafusion::scalar::ScalarValue::Utf8(Some(
-                "x".to_string(),
-            )))),
-        }));
+        let expr = (col("foo").eq(lit("x"))).not();
```
Comment (alamb) [inline predicate/src/delete_expr.rs:180]: these are the same expressions, but constructed using the "fluent style" expr API

File: query_tests/cases/in/pushdown.expected [hunk 3]
```diff
@@ -135,21 +135,23 @@
...
+|               |   Projection: restaurant.count, restaurant.system, restaurant.time, restaurant.town                                                                                                                                                                                                                                                   |
+|               |     Filter: CAST(restaurant.count AS Int64)restaurant.count > Int64(200) AND restaurant.town != Dictionary(Int32, Utf8("tewsbury")) AND restaurant.system = Float64(5) OR restaurant.town = Dictionary(Int32, Utf8("lawrence")) AND CAST(restaurant.count AS Int64)restaurant.count < Int64(40000)                                    |
+|               |       Projection: CAST(restaurant.count AS Int64) AS CAST(restaurant.count AS Int64)restaurant.count, restaurant.count, restaurant.system, restaurant.time, restaurant.town                                                                                                                                                           |
```
Comment (alamb) [inline query_tests/cases/in/pushdown.expected:144]: the diff doesn't make this easy to see, but the difference is that a projection has been pushed down below the filter which should not change anything

**Review by mkmik — APPROVED** [review]
Awesome

**Pull request merged**